gpflow/experimental/check_shapes/decorator.py

```diff
@@ -63,6 +63,10 @@ def check_shapes(*specs: str) -> Callable[[C], C]:
        :dedent:
 
     :param specs: Specification of arguments to check. See: `Check specification`_.
+    :param tf_decorator: Whether to wrap the shape check with
+        ``tf.compat.v1.flags.tf_decorator.make_decorator``.
+        Setting this `True` seems to solve some problems, particularly related to Keras models,
+        but create some other problems, particularly related to branching on tensors.
```

Comment:
This is a pretty vague description. Do we have a better understanding of the trade-offs?

Reply:
Not really. To me Tensorflow is big and complicated and fragile. You make a small change in one place and something else breaks somewhere else. It's called `compat.v1` so hopefully it's deprecated and will be going away...